File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/MultipleVariableDeclaration.vb
Namespace Tests.Diagnostics

    Public Class MultipleVariableDeclaration
        Public Const AAA As Integer = 5, 'some comment
            BBB = 42  ' Noncompliant {{Declare 'BBB' in a separate statement.}}
'           ^^^
        Public Const DDD As String = "foo"
        Public Sub New(n As Integer)
            Dim Const AAA, B As Integer = 5, ' Noncompliant
                BBB = 42, ' Noncompliant
                CCC As String = "foo"  ' Noncompliant
            Dim Const DDD As String = "foo"
        End Sub
    End Class
End Namespace

File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/MultipleVariableDeclaration.cs
namespace Tests.Diagnostics
{
    public class MultipleVariableDeclaration
    {
        private int /*aaa1*/ a /*aaa2*/,
            /*bbb1*/ b /*bbb2*/; // Noncompliant
//                   ^
        private int c;
        public MultipleVariableDeclaration(int n)
        {
            int /*aaa1*/a = 0 /*aaa2*/, /*bbb1*/ b = 0 /*bbb2*/; // Noncompliant {{Declare 'b' in a separate statement.}}
//                                               ^
            int c = 0;
            for (int i = 0, j = 2; i < 3; i++)
            {

            }
        }
    }
}
